Fix broken docs

aorenste opened the pull request:
These were causing doctest to be unhappy.

In particular the doc from #124496 caused #124771 to fail "trunk / win-vs2019-cpu-py3 / test" to fail when pushing. Not sure why it wasn't a problem on the original PR.

Testing:

`./test/run_doctests.sh`:
  before:
```
=== 4 warnings in 11.21 seconds ===
```
  after:
```
===  in 11.11 seconds ===
```

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #124940

aorenste committed:
Fix broken docs

[ghstack-poisoned]

zou3519 reviewed (approved):
interesting, thank you

aorenste committed:
Update on "Fix broken docs"

These were causing doctest to be unhappy.

In particular the doc from #124496 caused #124771 to fail "trunk / win-vs2019-cpu-py3 / test" to fail when pushing. Not sure why it wasn't a problem on the original PR.

Testing:

`./test/run_doctests.sh`:
  before:
```
=== 4 warnings in 11.21 seconds ===
```
  after:
```
===  in 11.11 seconds ===
```




[ghstack-poisoned]

zou3519 commented on torch/library.py:
```diff
@@ -853,35 +853,38 @@ def opcheck(
-        >>>     x_np = x.numpy(force=True)
-        >>>     z_np = x_np + y
-        >>>     return torch.from_numpy(z_np).to(x.device)
+        >>> def example():
```
Why do we need to put this into an example()?

aorenste replied:
With xdoctest v1.0.2 (which is what is installed on win-vs2019-cpu-py3) having the REQUIRES wrap a decorator seems to not work (it gets confused and complains about "key not found"). This seems to be fixed in later version of xdoctest - but wrapping in an example is a work-around.

zou3519 replied:
Can we just disable / ignore xdoctest on windows temporarily, or upgrade it? We don't want users to write code with this wrapping.

aorenste replied:
Let me see if I can figure out how to force a newer version...

aorenste committed:
Update on "Fix broken docs"

These were causing doctest to be unhappy.

In particular the doc from #124496 caused #124771 to fail "trunk / win-vs2019-cpu-py3 / test" to fail when pushing. Not sure why it wasn't a problem on the original PR.

Testing:

`./test/run_doctests.sh`:
  before:
```
=== 4 warnings in 11.21 seconds ===
```
  after:
```
===  in 11.11 seconds ===
```




[ghstack-poisoned]

aorenste committed:
Update on "Fix broken docs"

These were causing doctest to be unhappy.

In particular the doc from #124496 caused #124771 to fail "trunk / win-vs2019-cpu-py3 / test" to fail when pushing. Not sure why it wasn't a problem on the original PR.

Testing:

`./test/run_doctests.sh`:
  before:
```
=== 4 warnings in 11.21 seconds ===
```
  after:
```
===  in 11.11 seconds ===
```




[ghstack-poisoned]

huydhn reviewed (approved):
LGTM!  Just FYI, the list of Windows pip dependencies is here https://github.com/pytorch/test-infra/blob/main/aws/ami/windows/scripts/Installers/Install-Pip-Dependencies.ps1#L16, but we haven't updated it in a while because of the time taken to roll out a new Windows AMI.  

@atalman Do you think we should it's consolidate some few left over dependencies here before you roll out the new Windows AMI?  They should be safe to include because they are only CI dependencies

huydhn commented:
> @atalman Do you think we should it's consolidate some few left over dependencies here before you roll out the new Windows AMI? They should be safe to include because they are only CI dependencies

Oops, I'm too late, the new AMI is already there I https://github.com/pytorch-labs/pytorch-gha-infra/pull/382.  It's for the next time then

aorenste commented:
@pytorchbot merge
